### PR TITLE
Introduce video state bloat bond

### LINKED ISSUE #3394: Introduce video state bloat bond
# Dependency
https://github.com/Joystream/joystream/issues/3607

# Proposal
Currently missing, this means creating a video, without any storage assets, will generate no state bond, but it will take up space in storage. This is unlike channels, which have to have a storage bag created, so there is at least a proportional state cost, but it may be slightly too low, however that is acceptable. Fund with membership of caller, even if curator group or lead.

**Technical note:** As highlighted by @shamil-gadelshin, being able to introdue this requires the ability of the content pallet extrinsics to do balance sufficiency checks, before mutation, which covers the requirements of now itself and the storage directory, which requires a bond per data object. Hence some pallet method, like `total_bloat_bond_required(...) -> Balane`, has to be introduced that returns the cost required for the storage pallet. The content pallet can then add this to its own costs, and do a total balance check before it starts any mutation.

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201958687417145/1202127938101655) by [Unito](https://www.unito.io)

### PR DESCRIPTION
**Pull request by L3pereira**
* Closes #3394

Merge after #3821 

When creating a video, the sender gets slashed video state bloat bond
When deleting a channel sender gets paid the video state bloat bond amount
Uses the funds_needed_for_upload implemented here https://github.com/Joystream/joystream/pull/3693 to verify the funds needed before charging the video state bloat bond

┆Issue is synchronized with this [Asana task](https://app.asana.com/0/1201958687417145/1202478162412857) by [Unito](https://www.unito.io)

### CODE REVIEW

File: runtime-modules/content/src/lib.rs [hunk 1]
```diff
@@ -869,6 +916,8 @@ decl_module! {
             // == MUTATION SAFE ==
             //
 
+            let _ = Balances::<T>::slash(&sender, video_state_bloat_bond);
```
Comment (bedeho) [inline runtime-modules/content/src/lib.rs:879]: Do we not put the bloat bond in some treasury account or something? What did we do for bounty pallet bloat bonds, or for storage?

File: runtime-modules/content/src/lib.rs [hunk 2]
```diff
@@ -1185,6 +1240,36 @@ decl_module! {
...
+
+        /// Updates video state bloat bond value.
+        /// Only lead can upload this value
+        #[weight = 10_000_000] // TODO: adjust weight
+        pub fn update_video_state_bloat_bond(
+            origin,
+            new_video_state_bloat_bond: BalanceOf<T>,
+        ) {
+            let sender = ensure_signed(origin)?;
+            ensure_authorized_to_update_video_state_bloat_bond::<T>(&sender)?;
```
Comment (dobertRowneySr) [inline runtime-modules/content/src/lib.rs:1313]:
```suggestion
            ensure_authorized_to_update_video_state_bloat_bond::<T>(&sender)?;
            // == MUTATION SAFE ==
```

File: runtime-modules/content/src/tests/videos.rs [hunk 3]
```diff
@@ -39,12 +39,27 @@ fn successful_video_creation_by_member() {
...
-        increase_account_balance_helper(DEFAULT_MEMBER_ACCOUNT_ID, INITIAL_BALANCE);
+        //0 channel SBB
+        //+ 11 video SBB
+        //+ 0 channel data obj SBB * 10 objs
+        //+ 1 video data obj SBB * 10 objs
+        //= 11 SBB
+        increase_account_balance_helper(DEFAULT_MEMBER_ACCOUNT_ID, 11);
         create_default_member_owned_channel();
 
+        let video_state_bloat_bond = 1;
```
Comment (dobertRowneySr) [inline runtime-modules/content/src/tests/videos.rs:50]: why not use their `DEFAULT-*` counterparts?

**Review by dobertRowneySr — CHANGES_REQUESTED** [review]
Same cosmetics as the channel bloat bond PR, otherwise LGTM

**Review by dobertRowneySr — APPROVED** [review]
LGTM

**Review by dobertRowneySr — APPROVED** [review]

**Review by dobertRowneySr — CHANGES_REQUESTED** [review]
- build node & runtime :heavy_check_mark: 
- types and metadata :heavy_check_mark: 
- cargo test :heavy_check_mark: 
- there are some problem with benchmarks: 
```
---- tests::council_discard_remaining_rewards_on_depose stdout ----
thread 'tests::council_discard_remaining_rewards_on_depose' panicked at 'attempt to subtract with overflow', runtime-modules/council/src/mock.rs:1496:17
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace

---- tests::council_many_cycle_rewards stdout ----
thread 'tests::council_many_cycle_rewards' panicked at 'attempt to subtract with overflow', runtime-modules/council/src/mock.rs:1496:17

---- tests::council_rewards_are_paid stdout ----
thread 'tests::council_rewards_are_paid' panicked at 'attempt to subtract with overflow', runtime-modules/council/src/mock.rs:1496:17

---- tests::vote_stake_locks_after_new_election_starts stdout ----
thread 'tests::vote_stake_locks_after_new_election_starts' panicked at 'attempt to subtract with overflow', runtime-modules/council/src/mock.rs:1496:17
```
These should be fixed see https://github.com/mnaamani/joystream/pull/13

**Review by dobertRowneySr — APPROVED** [review]

**Pull request merged**